src/analysis/inner_product_space/finite_dimensional.lean

```diff
@@ -0,0 +1,537 @@
+/-
+Copyright (c) 2023 Monica Omar. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Monica Omar
+-/
+import analysis.inner_product_space.adjoint
+import analysis.inner_product_space.spectrum
+
+/-!
+# Finite-dimensional inner product spaces
+
+In this file, we prove some results in finite-dimensional inner product spaces.
+
+## Notation
+
+This file uses the local notation `P _` for `orthogonal_projection _`
+and `↥P _` for the extended orthogonal projection `V →L[ℂ] V`.
+
+We let `V` be an inner product space over `ℂ`.
+-/
+
+variables {V : Type*} [inner_product_space ℂ V]
+local notation `P` := orthogonal_projection
+
+/-- `U` is `T` invariant: `∀ u : V`, if `u ∈ U` then `T u ∈ U`-/
+def submodule.invariant (U : submodule ℂ V) (T : V →ₗ[ℂ] V) : Prop := U ≤ U.comap T
+lemma submodule.invariant_iff (U : submodule ℂ V) (T : V →ₗ[ℂ] V) :
+  submodule.invariant U T ↔ T '' U ⊆ U :=
+by simp only [set.image_subset_iff]; refl
+
+/-- `↥P _` is the extended version of `P _` -/
+noncomputable def orthogonal_projection.extend [finite_dimensional ℂ V]
```

Comment:
This makes sense both over the reals and the complex numbers, so it should be stated for a `is_R_or_C` field. And since this has nothing specific about finite dimensional, it should probably be in a different file (likely the one where the orthogonal projection is defined). (The right assumption is that `U` is complete, just like for `orthogonal_projection`).